test/test_test.c

```diff
@@ -22,28 +22,34 @@
 #include "test_main.h"
 #include "testutil.h"
 
-#define C(l, b, t)                                      \
-    if ((t) != b) {                                     \
-        fprintf(stderr, "FATAL : %s != %d\n", #t, b);   \
-        goto l;                                         \
+#define TEST(b, t) test_case(#t, (t), (b))
```

Comment:
Does it have to flip order? Do parameters have to be just one letter? It still looks like "if there is something one can do to makes it harder to read, then we do it." Yes, test_case() is right in next line, but it kind of makes it worse, as you have to strain to map b-to-the-left to expected-to-the-right...